src/api/Movies/Movies.api.ts

```diff
@@ -0,0 +1,49 @@
+import axios from 'axios';
+
+import { API_URL } from './api.constants';
+import {
+  Movie,
+  MovieApi,
+  MovieApiResponse,
+  FetchMovieParams,
+} from './Movie.models';
+
+export const fetchMoviesAPI = async (
+  params: FetchMovieParams
+): Promise<MovieApiResponse<Movie[]>> => {
+  const apiResponse = await axios.get<MovieApiResponse<MovieApi[]>>(
+    `${API_URL}/movies`,
+    {
+      params,
+      paramsSerializer(param: Record<string, any>) {
```

Comment:
This particular logic may be required in multiple places across your app, so, probably, would make sense to abstract that away as a separate helper method.

Other than that, I would leverage [axios instance](https://axios-http.com/docs/instance) here and decorate that within different data fetching models as needed.